src/dftbp/timedep/timeprop.F90

```diff
@@ -2234,6 +2236,14 @@ subroutine initTDOutput(this, dipoleDat, qDat, energyDat, populDat, forceDat, co
       end do
     end if
 
+    iErr = -999
+    if (this%tPump) then
+      call execute_command_line("mkdir "//trim(pumpFilesDir), exitstat=iErr)
```

Comment:
Can't you call into the C library instead of resorting to a shell escape here?

Reply:
I don't know how to use the C library. Do you have an example in the code where you mkdir using this library? This is how it's done in the Poisson code

Reply:
See

- https://www.gnu.org/software/libc/manual/html_node/Creating-Directories.html
- https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/mkdir-wmkdir?view=msvc-170

Reply:
Do we have a Fortran interface to this?

Reply:
No, not yet. However, I definitely would like to avoid to have a shell-call within the code. (Poisson does not count, it is declared as "external library", meaning, we don't take responsibility, how things are coded up there :wink: ). libNEGF uses something more appopriate (see `ext_system/system_calls.f90`), but that's deep within libNEGF...

I have actually functional wrappers for almost all file manipulation routines of libc, I just have to set up a proper CMake build system for it. I won't manage it, though, before the release. (fragments of some earlier projects of mine....)

Reply:
Sorry, I did not realise that this would be that complicated! Let's forget about this PR in that case. I'll add the mkdir and move files to the Python scripts and do it as a postprocessing.